pkg/middleware/stored_session.go

```diff
@@ -73,7 +73,10 @@ func (s *storedSessionLoader) loadSession(next http.Handler) http.Handler {
 		if err != nil {
 			// In the case when there was an error loading the session,
 			// we should clear the session
-			logger.Errorf("Error loading cookied session: %v, removing session", err)
+			if scope.SessionRequired {
+				// log error because session is required but we didn't find one
+				logger.Errorf("Error loading cookied session: %v, removing session", err)
+			}
```

Comment:
I don't think this is correct, if there's an invalid session, we should still be logging this message. If no session is found, then no error is returned so this won't be logged will it?

Reply:
The logic here is that `scope.SessionRequired` being `true` essentially means "the current route is _not_ skipped" - for an auth-required route, there being no cookie should be logged as an error, but not for a skipped route.

This is the crux of what I'm trying to address with this PR - for skipped routes the absence of a session shouldn't be an error, but _if_ there is a session available then we want to inject the headers when calling upstream the same as we would for authentication-required routes.  To do this we have to _attempt_ to build the session as we would with authentication required, and just keep quiet if we can't.

Reply:
Maybe the code would make more sense if I reversed the sense of the flag to make it

```
if !scope.SessionOptional {
```